transformer_engine/pytorch/fp8.py

```diff
@@ -314,7 +314,7 @@ def _default_get_amax(
     if amax_compute_algo == "max":
         amax = torch.max(amax_history, dim=0).values
     else:  # amax_compute_algo == "most_recent"
-        amax = amax_history[0]
+        amax = amax_history[0].clone()
```

Comment:
In the `update_amax_history` call that follows, for the case with no amax history we avoid the roll kernel, but as a result we also overwrite this amax when calling `fill_` on the tensor. Making a copy avoids this.